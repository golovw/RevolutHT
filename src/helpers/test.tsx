import React from 'react';
import { render } from '@testing-library/react';
import { ThemeProvider } from 'styled-components';

import theme from 'styles';

export const wrapWithTheme = (Component: React.ReactNode) => (
  <ThemeProvider theme={theme}>
    {Component}
  </ThemeProvider>
);

export const renderWithTheme = (Component: React.ReactNode, options?: any) => render(wrapWithTheme(Component), options);
