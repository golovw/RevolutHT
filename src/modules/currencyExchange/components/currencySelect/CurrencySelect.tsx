import React from 'react';

import MenuItem from '@material-ui/core/MenuItem';

import { Account as AccountI } from 'ducks/Account.duck';

import { Select } from './CurrencySelect.styles';

interface Props {
  value: string;
  accounts: AccountI[];
  onChange?: (event: React.ChangeEvent<{ value: unknown }>) => void;
}

const renderMenuItem = ({ id, currency }: AccountI) => (
  <MenuItem key={id} value={id}>{currency}</MenuItem>
);

export const CurrencySelect = (props: Props) => {
  const {
    value,
    accounts,
    onChange,
  } = props;

  return (
    <Select disableUnderline value={value} onChange={onChange}>
      {accounts.map(renderMenuItem)}
    </Select>
  );
};
