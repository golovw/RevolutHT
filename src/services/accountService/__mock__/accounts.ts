export const accountsMock = [
  {
    id: '33bba04a',
    currency: 'USD',
    balance: '2340.00',
  },
  {
    id: '33bba31a',
    currency: 'EUR',
    balance: '87.00'
  },
  {
    id: '33bba428',
    currency: 'GBP',
    balance: '104.00'
  },
];
