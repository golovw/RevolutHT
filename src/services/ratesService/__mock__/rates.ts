export const ratesMock = {
  EUR: 0.818746,
  GBP: 0.706283,
  USD: 1,
};
