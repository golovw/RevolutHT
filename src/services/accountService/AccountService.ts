import { accountsMock } from './__mock__';

import { ApiAdapterI } from 'adapters/apiAdapter';

export class AccountService {
  apiAdapter;

  constructor(apiAdapter: ApiAdapterI) {
    this.apiAdapter = apiAdapter;
  }

  getAccounts = async () => {
    return { data: accountsMock };
  }
}
