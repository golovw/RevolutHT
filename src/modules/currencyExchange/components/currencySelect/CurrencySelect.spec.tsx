import React from 'react';

import { renderWithTheme } from 'helpers/test';

import { CurrencySelect } from './CurrencySelect';

import { accountsMock } from 'services/accountService/__mock__';

describe('CurrencySelect', () => {

  describe('Snapshots', () => {
    it('currency select should match snapshot', () => {
      const component = renderWithTheme(
        <CurrencySelect
          value={accountsMock[0].id}
          accounts={accountsMock}
          onChange={() => {}}
        />
      );

      expect(component).toMatchSnapshot();
    });
  });

});
