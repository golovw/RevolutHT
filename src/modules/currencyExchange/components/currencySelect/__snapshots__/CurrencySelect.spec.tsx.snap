// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CurrencySelect Snapshots currency select should match snapshot 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="MuiInputBase-root MuiInput-root sc-bdnxRM eMbhKg"
      >
        <div
          aria-haspopup="listbox"
          class="MuiSelect-root MuiSelect-select MuiSelect-selectMenu MuiInputBase-input MuiInput-input"
          role="button"
          tabindex="0"
        >
          USD
        </div>
        <input
          aria-hidden="true"
          class="MuiSelect-nativeInput"
          tabindex="-1"
          value="33bba04a"
        />
        <svg
          aria-hidden="true"
          class="MuiSvgIcon-root MuiSelect-icon"
          focusable="false"
          viewBox="0 0 24 24"
        >
          <path
            d="M7 10l5 5 5-5z"
          />
        </svg>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="MuiInputBase-root MuiInput-root sc-bdnxRM eMbhKg"
    >
      <div
        aria-haspopup="listbox"
        class="MuiSelect-root MuiSelect-select MuiSelect-selectMenu MuiInputBase-input MuiInput-input"
        role="button"
        tabindex="0"
      >
        USD
      </div>
      <input
        aria-hidden="true"
        class="MuiSelect-nativeInput"
        tabindex="-1"
        value="33bba04a"
      />
      <svg
        aria-hidden="true"
        class="MuiSvgIcon-root MuiSelect-icon"
        focusable="false"
        viewBox="0 0 24 24"
      >
        <path
          d="M7 10l5 5 5-5z"
        />
      </svg>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
